torch/csrc/jit/codegen/cuda/scheduler/vectorize_helper.h

```diff
@@ -1,37 +1,487 @@
+    }
+
+    valid_extents_ = false;
+    is_simplified_ = false;
+  }
+
+  // If necessary simplify the PartialExtent and return all the numerator values
+  // multiplied together as a single Val.
+  Val* getNumerator() {
+    if (!valid_extents_) {
+      computeNumerDenomir();
+    }
+    return numerator_;
+  }
+
+  // If necessary simplify the PartialExtent and return all the denominator
+  // values multiplied together as a single Val.
+  Val* getDenominator() {
+    if (!valid_extents_) {
+      computeNumerDenomir();
+    }
+    return denominator_;
+  }
+
+  bool isZero() const {
+    return zero_;
+  }
+
+  std::string toString() const {
+    std::stringstream ss;
+    ss << "{";
+    bool first = true;
+    for (auto numer : const_numerator_vals_) {
+      if (!first) {
+        ss << " * ";
+      }
+      first = false;
+
+      ss << numer;
+    }
+
+    for (auto numer : symb_numerator_vals_) {
+      if (numer == nullptr || numer->isOneInt()) {
+        continue;
+      }
+      if (!first) {
+        ss << " * ";
+      }
+      first = false;
+      ss << numer->toInlineString();
+    }
+
+    ss << "} / {";
+    first = true;
+    for (auto denom : const_denominator_vals_) {
+      if (!first) {
+        ss << " * ";
+      }
+      first = false;
+
+      ss << denom;
+    }
+
+    for (auto denom : symb_denominator_vals_) {
+      if (denom == nullptr || denom->isOneInt()) {
+        continue;
+      }
+      if (!first) {
+        ss << " * ";
+      }
+      first = false;
+
+      ss << denom->toInlineString();
+    }
+    ss << "}";
+    return ss.str();
+  }
+
+ private:
+  // Simplify the partial extent as much as possibly by removing common factors
+  // from the numerator and denominator.
+  void simplify() {
+    if (is_simplified_) {
+      return;
+    }
+
+    std::tie(const_numerator_vals_, const_denominator_vals_) =
+        factorization_helpers::removeCommonFactors(
+            const_numerator_vals_, const_denominator_vals_);
+
+    std::tie(symb_numerator_vals_, symb_denominator_vals_) =
+        factorization_helpers::removeSameVals(
+            symb_numerator_vals_, symb_denominator_vals_);
+
+    valid_extents_ = false;
+    is_simplified_ = true;
+  }
+
+  // Simplify the partial extent and generate the Val* for numerator and
+  // denominator by multiplying all the factors.
+  void computeNumerDenomir() {
+    if (valid_extents_) {
+      return;
+    }
+    simplify();
+
+    numerator_ = nullptr;
+
+    auto const_numerator_factor = 1;
+    for (auto factor : const_numerator_vals_) {
+      const_numerator_factor *= factor;
+    }
+
+    if (const_numerator_factor > 1) {
+      numerator_ = IrBuilder::create<Int>(const_numerator_factor);
+    }
+
+    for (auto numerator_val : symb_numerator_vals_) {
+      if (numerator_ == nullptr) {
+        numerator_ = numerator_val;
+      } else {
+        numerator_ = IrBuilder::mulExpr(numerator_, numerator_val);
```

Comment:
Can we just use `SimplifyingIrBuilder`?

Reply:
Yeah, I'm not sure why we wouldn't just always use simplifying ir builder, except that it might be harder when we read the code to know where the constant value comes from. We probably should add a disable option to disable simplifying IR expression.

Reply:
Went through and tried to update all uses to IrBuilder that made sense to move to simplifyingIrBuilder